packages/@aws-cdk/aws-apigatewayv2/lib/http/integrations/private-integration/application-loadbalancer-integration.ts

```diff
@@ -0,0 +1,25 @@
+import * as elbv2 from '@aws-cdk/aws-elasticloadbalancingv2';
+import { BasePrivateIntegrationProps, HttpProxyPrivateIntegration } from './http-private';
+
+/**
+ * Properties to initialize `ApplicationLoadBalancerPrivateIntegration`.
+ */
+export interface ApplicationLoadBalancerPrivateIntegrationProps extends BasePrivateIntegrationProps {
+  /**
+   * The listener to the application load balancer used for the integration
+   */
+  readonly listener: elbv2.IApplicationListener;
+}
+
+/**
+ * The Application Load Balancer Private integration resource for HTTP API
+ */
+export class ApplicationLoadBalancerPrivateIntegration extends HttpProxyPrivateIntegration {
```

Comment:
Some terser names would be nicer.

```suggestion
export class ALBIntegration extends HttpProxyPrivateIntegration {
```

Reply:
Done!